samples/testing-frameworks/appium/client-side/csharp/biometric-sample-test/BitbarBiometricSample/TestBiometricsAndroid.cs

```diff
@@ -0,0 +1,79 @@
+using OpenQA.Selenium.Appium.Android;
+
+namespace BitbarBiometricSample
+{
+    [TestFixture]
+    public class TestBiometricsAndroid
+    {
+        private readonly String SCREENSHOT_DIRECTORY = AppDomain.CurrentDomain.BaseDirectory + "/screenshots/";
+        private RemoteWebDriver driver;
+
+        [OneTimeSetUp]
+        public void InitDriver()
+        {
+            const String AppiumHubURL = "https://appium.bitbar.com/wd/hub";
+
+            AppiumOptions capabilities = new();
+            //Platform dependent
+            capabilities.AddAdditionalCapability("platformName", "Android");
+            capabilities.AddAdditionalCapability("deviceName", "Android Phone");
+            capabilities.AddAdditionalCapability("automationName", "Appium");
+            //Customizable
+            capabilities.AddAdditionalCapability("bitbar_project", "C# Appium Automated Test");
+            capabilities.AddAdditionalCapability("bitbar_testrun", "Android Run");
+            capabilities.AddAdditionalCapability("bitbar_device", "Google Pixel 2");
+            //Required
+            capabilities.AddAdditionalCapability("bitbar_apiKey", "<YOUR_BITBAR_API_KEY>");
+            capabilities.AddAdditionalCapability("bitbar_app", "<APP_ID>");
+            //Biometrics capability
+            capabilities.AddAdditionalCapability("biometricInstrumentation", true);
+
+            driver = new AndroidDriver<AndroidElement>(new Uri(AppiumHubURL), capabilities, TimeSpan.FromSeconds(300));
+        }
+
+        [OneTimeTearDown]
+        public void QuitDriver()
+        {
+            driver.Quit();
+        }
+
+        [Test]
+        public void Test_Biometric_Authentication()
+        {
+            Console.WriteLine("Test execution started:");
+
+            TakeScreenshot("HomeScreen");
+            var buttonBiometrics = driver.FindElement(By.XPath("//android.view.ViewGroup[@content-desc=\"Biometric authentication\"]"));
+            buttonBiometrics.Click();
+
+            var textSensorType = driver.FindElement(MobileBy.AccessibilityId("Sensor type value"));
+            Assert.That(textSensorType.Text.ToUpper, Is.EqualTo("ANDROID BIOMETRICS"));
+
+            var buttonAskBiometrics = driver.FindElement(By.XPath("//android.view.ViewGroup[@content-desc=\"Ask biometric authentication\"]"));
+            buttonAskBiometrics.Click();
+            TakeScreenshot("Dialog");
+
+            var buttonPassBiometrics = driver.FindElement(By.XPath("//*[@text=\"PASS\"]"));
+            buttonPassBiometrics.Click();
+
+            var textResult = driver.FindElement(MobileBy.AccessibilityId("Authentication status value"));
+            TakeScreenshot("Result");
+            Assert.That(textResult.Text.ToUpper, Is.EqualTo("SUCCEEDED"));
+        }
+
+        public void TakeScreenshot(String fileName)
```

Comment:
I think we should extract it to another class. That code is duplicated in the second test file